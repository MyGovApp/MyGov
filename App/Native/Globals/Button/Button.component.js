import React, { Component, PropTypes } from 'react'
import { Text, TouchableOpacity } from 'react-native'
import s from './Button.styles'

class Button extends Component {
  render () {
    const { text, onPress, textStyle, buttonStyle } = this.props
    return (
      <TouchableOpacity
        onPress={onPress}
        style={[ s.default.container, buttonStyle ]}
      >
        <Text
          style={[ s.default.text, textStyle ]}
        >{text}</Text>
      </TouchableOpacity>
    )
  }
}

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  textStyle: PropTypes.object,
  buttonStyle: PropTypes.object
}

export default Button
