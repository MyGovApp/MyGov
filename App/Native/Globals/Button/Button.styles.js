import { fonts, colors } from '../../Themes'

export default {
  default: {
    container: {
      backgroundColor: colors.primaryBlue,
      borderRadius: 4,
      paddingTop: 10,
      paddingBottom: 10,
      paddingRight: 20,
      paddingLeft: 20
    },
    text: {
      ...fonts.style.normal,
      color: 'white'
    }
  }
}
