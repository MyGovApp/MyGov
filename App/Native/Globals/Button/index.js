import Button from './Button.component.js'

export default Button
