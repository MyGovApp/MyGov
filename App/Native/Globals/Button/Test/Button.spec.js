import 'react-native'
import React from 'react' // eslint-disable-line
import renderer from 'react-test-renderer'
import { ComponentWrapper } from './Button.setup'

describe('Native Hello World', () => {
  it('should render the Button component correctly', () => {
    const tree = renderer.create(ComponentWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
