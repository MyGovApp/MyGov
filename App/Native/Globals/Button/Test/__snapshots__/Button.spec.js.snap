// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Native Hello World should render the Button component correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#112E50",
      "borderRadius": 4,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 10,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "HelveticaNeue",
          "fontSize": 17,
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
